Change camera state to enum (#2330)
home-assistant/developers.home-assistant: blog/2024-09-23-state-constants-camera-deprecation.md

---
author: G Johansson
authorURL: https://github.com/gjohansson-ST
authorImageURL: https://avatars.githubusercontent.com/u/62932417?v=4
authorTwitter: GJohansson
title: "Deprecating state constants for camera"
---

> As of Home Assistant Core 2024.10, the constants used as returning state in `Camera` is deprecated and replaced by the `CameraState` enum.

Reviewer: ```suggestion
As of Home Assistant Core 2024.10, the constants used as returning state in `Camera` are deprecated and replaced by the `CameraState` enum.
```

> As of Home Assistant Core 2024.10, the constants used as returning state in `Camera` are deprecated and replaced by the `CameraState` enum.

Author: ```suggestion
As of Home Assistant Core 2024.10, the constants used to return state in `Camera` are deprecated and replaced by the `CameraState` enum.
```

> There is a one-year deprecation period, and the the constants will stop to work from 2025.10 to ensure all custom integration authors have time to adjust.

Reviewer: ```suggestion
There is a one-year deprecation period, and the constants will stop working from 2025.10 to ensure all custom integration authors have time to adjust.
```

> As the `state` property is not meant to be overwritten, in most cases this is only related for use in other properties or in tests.

Reviewer: ```suggestion
As the `state` property is not meant to be overwritten, in most cases this change will only affect other Entity properties or tests.
```
